### ISSUE TITLE
Home icon set with ground, 1st and attic floors for home automation

### ISSUE THREAD

**Issue by james-fry** [Templarian/MaterialDesign#1964]
Hi,

I designed a set of icons to represent the floors of a home (ground, 1st and attic) for home automation purposes (e.g. group/view for different floors, sensors/switches on floors etc).
I created a new simpler home icon first as I feel it is neater to show the floors this way, but I am also submitting floors versions of the default solid home icon from MDI.

[home-simple.zip](https://github.com/Templarian/MaterialDesign/files/870465/home-simple.zip)
[home.zip](https://github.com/Templarian/MaterialDesign/files/870466/home.zip)

**Comment by MrGrigri** [comment]
I really like the `home-simple` ones. I believe that they are less confusing and more on-par with MD.

**Comment by james-fry** [comment]
The home-simple are my preference also.

**Comment by rtvb** [comment]
I like the idea, too. And yes, the home-simple set is a better fit. However they look a little similar to the current garage icons. 

I would additionally like a basement or cellar icon.

**Comment by MrGrigri** [comment]
@rtvb, even though they look similar, they function and their identification is dictated by it's surrounding. It's up to the application designer to implement icons in a way that there is no confusion that they want it to be a level in a house instead of a garage door.

**Comment by james-fry** [comment]
@rtvb I agree they look similar to the garage, but used for different purposes. Also there are not enough variations of garage to fulfill the objective of indicating floors in a home.
Regarding basement, I agree it would be nice but to do it properly (i.e. keeping good spacing etc) I would need to redesign all the other icons. I attached the best I can do without the refactoring.
[home-simple-basement.zip](https://github.com/Templarian/MaterialDesign/files/888681/home-simple-basement.zip)

**Comment by james-fry** [comment]
Please can this post be labelled with "Home Assistant".
Also is there any info on how the submissions process works? There are of course comments etc in this thread, but what is the decision process on if/when icons will be added to main? Or what changes would need to be made to meet quality criteria?

**Comment by james-fry** [comment]
Is there anything further that would need to be done to get these icons added?

**Comment by JapanYoshi** [comment]
Please remember that the US and UK have different names for these floors. I suggest the names ``home-simple-floor-low`` and ``home-simple-floor-high``.

**Comment by james-fry** [comment]
Its a good point - thanks.
I updated the names to be locale agnostic per your suggestion.

The icons are maintained here:
https://github.com/james-fry/svg_icons/tree/master/home-simple

[home-simple.zip](https://github.com/Templarian/MaterialDesign/files/1212095/home-simple.zip)

**Comment by james-fry** [comment]
Seems there is more demand from the Home Assistant community for icons to represent floors in a home. 
https://community.home-assistant.io/t/custom-icons-top-view/29928
What needs to done to progress this icon submission?

**Comment by brent20** [comment]
Any updates on getting these icons merged?

**Comment by CoDEmanX** [comment]
Previews:

![image](https://user-images.githubusercontent.com/1309249/38770659-ac360b04-4016-11e8-859b-b56353b0a0c6.png) ![image](https://user-images.githubusercontent.com/1309249/38770669-d4101c5a-4016-11e8-89ab-bc96363c875c.png)
![image](https://user-images.githubusercontent.com/1309249/38770674-e9e0630a-4016-11e8-833f-08f13617c1b8.png) ![image](https://user-images.githubusercontent.com/1309249/38770683-215446f8-4017-11e8-9ed8-9a56d98c86c8.png)
![image](https://user-images.githubusercontent.com/1309249/38770686-336a74a2-4017-11e8-97b4-66de5bb8dd46.png) ![image](https://user-images.githubusercontent.com/1309249/38770688-3ef7b3fc-4017-11e8-91ce-cc073ce22b52.png)

At larger sizes, the floor and attic ones look a bit odd. I think it's the lack of rounded corners and the spacing. It could work with a couple of tweaks though. The basement variant should be moved down 1px to be consistent.

**Comment by james-fry** [comment]
I'd be happy to make the changes, but I'm not 100% sure what you're requesting re rounded corners and spacing - do you mean on the floor indicators within the house shape?
Re the basement icon, this was added at another users request and I agree it's not nicely consistent with the other floors, but I couldnt get it to work any better... please by all means feel free to make any changes :)

**Comment by CoDEmanX** [comment]
I'm not sure either. I think it's the spacing of the second floor. But I don't like it this way either:

![grafik](https://user-images.githubusercontent.com/1309249/38839598-b72d4a1a-41db-11e8-84d5-4c9690339d77.png)

Inverting kinda works for all floors, but not sure about the other configuration with only one floor highlighted...

![grafik](https://user-images.githubusercontent.com/1309249/38839779-94bcf290-41dc-11e8-8bfc-d078409e1b17.png)

(corners have 2px rounding BTW., as a test)

**Comment by GreenTurtwig** [comment]
![](https://user-images.githubusercontent.com/15905797/38867571-4cdd06e4-423c-11e8-8d8a-9855e8f24710.png)![](https://user-images.githubusercontent.com/15905797/38867621-6bc4be62-423c-11e8-96f9-6d26740c447b.png)![](https://user-images.githubusercontent.com/15905797/38867521-331cb664-423c-11e8-838a-97f3fcabb7de.png)![](https://user-images.githubusercontent.com/15905797/38867651-850aa1ca-423c-11e8-81e4-6110d52bea63.png)

If they were consistent with the normal `home` icons they'd look like this.

**Comment by GreenTurtwig** [comment]
![](https://user-images.githubusercontent.com/15905797/38868105-e5c900be-423d-11e8-80d0-98f754dacf81.png)![](https://user-images.githubusercontent.com/15905797/38868134-f4aaf1b4-423d-11e8-8f2f-1a7f308c5ed4.png)![](https://user-images.githubusercontent.com/15905797/38868146-045637ea-423e-11e8-8dc3-c4a0274ec996.png)![](https://user-images.githubusercontent.com/15905797/38868166-12871e56-423e-11e8-9b95-bb1d81a68490.png)

Inverted them for the hell of it.

**Comment by james-fry** [comment]
Thanks for more options!
My personal preference is for the simple home as I captured it - I just prefer that google home style.
i.e.
https://lh3.googleusercontent.com/EIX5pscg5l1Quvsz17LKk6W5oCSIxOfukBAUmbGR4Uvs1uKDYvbNr0XYE7hqpI3KHQ=s180

Regarding the alphanumeric for the home floor/position - great idea. It allows more flexibility for home size (3 floors) but localisation becomes important (attic vs loft, UK Ground = US 1 etc).
I support the above icons as the most flexible and consistent (with Google MD home icon) approach (even though my personal aesthetic preference is for the outline home icon that is based on google home)

**Comment by PeterShaggyNoble** [comment]
Attic, as discussed on Slack, is going to be a tricky one as, whichever you refer to it as, neither `A` nor `L` are going to be understood in other languages.

**Comment by james-fry** [comment]
Same is true for B, although usage for elevator (lift :)) is probably well understood.
But you have to draw the line somewhere...
Maybe if there are a 0, 3 and 4 versions there are enough permutations to allow people to substitute e.g. 3 for attic.

**Comment by GreenTurtwig** [comment]
Yep, both basement and ground letters are understood in most parts of the world. We would likely have to rename `floor-loft` to `floor-lobby` though as that's what the letter usually represents.

**Comment by Templarian** [comment]
I'm thinking of adding the ones with the numbers. I'm still not sure what we should do about the letters.

We've been trying very hard to keep things as international as possible, so maybe there is a nice way to represent this?

**Comment by mueller-ma** [comment]
> international as possible

Numbers aren't really international as well. See https://en.wikipedia.org/wiki/Storey#Consecutive_number_floor_designations

**Comment by Templarian** [comment]
That's up to the application to determine though. They are still usable. The issue with the letters is there would be no solution for other languages.

**Comment by johntdyer** [comment]
Really looking forward to this being merged

**Comment by goyney** [comment]
I think we should add variations of the home with the numbers 0 thru 3 or 4. This would allow us to accommodate calling the ground floor 0 or 1 and typically residential homes don't really go over 3 stories.

If we want to go all out, we *could* do 0-9 to match the `numeric` sets, but I don't know if that is really needed.

**Comment by GreenTurtwig** [comment]
@goyney I agree, although I'd add -1 to that list for basements.

**Comment by Templarian** [comment]
I'm going to reassign this to the next milestone which will be going out shortly. I think we should just go with the numbers/letters and let users assign where appropriate.

Assuming 3 stories etc seems too limiting.

**Comment by james-fry** [comment]
Great. Looking forward to it @Templarian. thanks

**Comment by Templarian** [comment]
@Roemer Sure, can you please create another issue and I'll add it to the `3.5.xx` milestone we just started. Should be pretty quick.
